R/query_dataset.R

```diff
@@ -0,0 +1,37 @@
+#' Query a data set using GET and a query URL
+#'
+#' @description
+#' This function provides a method for generating and sending a URL based data query to the
+#' EES API. As a minimum, it requires the dataset_id and indicators flags to be provided.
+#'
+#' @param dataset_id ID of data set to be connected to
+#' @param indicators Indicators required as a string or vector of strings
+#' @param time_periods Time periods required as a string ("period|code") or vector of strings
+#' @param geographic_levels Geographic levels required as a string or vector of strings
+#' @param dataset_version Version of data set to be connected to
+#' @param api_version EES API version
+#'
+#' @return Data frame containing query results of an API data set
+#' @export
+#'
+#' @examples
+#' query_dataset_get(example_id(), indicators = "session_count")
+query_dataset_get <- function(
```

Comment:
I can see what you're doing with the names here, as you'll have a `query_dataset.*` family. The word order hurts my a head a little and I think I'd read them easier if we had something like:

`get_dataset()`
`post_dataset_query()`

This differentiates the post endpoint wrapper a bit but that's no bad thing as ultimately that's where we do want to push most users towards